torch/csrc/autograd/variable.h

```diff
@@ -248,12 +245,25 @@ struct TORCH_API AutogradMeta : public c10::AutogradMetaInterface {
     // set_requires_grad also checks error conditions.
     if (requires_grad) {
       TORCH_INTERNAL_ASSERT(self_impl);
-      set_requires_grad(requires_grad, self_impl);
+      set_requires_grad_base(requires_grad, self_impl);
     }
     TORCH_CHECK(
         !grad_fn_ || !requires_grad_,
         "requires_grad should be false if grad_fn is set");
   }
+
+private:
+  // This base method must be separate from the virtual method
+  // `set_requires_grad` so that the constructor can call it directly. Calling
+  // a virtual function in a constructor will call the base override anyway,
+  // but it also causes a clang-tidy error. Calling it directly prevents the
+  // error.
+  void set_requires_grad_base(bool requires_grad, at::TensorImpl* self_impl) {
```

Comment:
I'm not sure why, but I was getting clang-tidy errors here, even though I wasn't changing any of the surrounding code. I can revert these changes if needed.